docs/azure-openshift.md

```diff
@@ -0,0 +1,165 @@
+[]
+```
+Initially you will see nothing. So you will have create a zone for your resourcegroup.
+
+7. Create a dns zone for you resourcegroup.
+```bash
+ az network dns zone create -g <azure_resourcegroup> -n <pick up the zone name from the route  eg- the part after apps. eg - misalunk-azure.qe.azure.devcluster.openshift.com>
+```
+
+8. Then create CR as follows -
+```yaml
+apiVersion: externaldns.olm.openshift.io/v1alpha1
+kind: ExternalDNS
+metadata:
+  name: sample-azure
+spec:
+  domains:
+  - filterType: Include
+    matchType: Exact
+    name: test-azure1.qe.azure.devcluster.openshift.com
+  provider:
+    type: Azure
+  source:
+    # Incase you have multiple ingress controllers you must specify ingress controller name in the routerName
+    # so that the external dns will use the router canonical name correrponding to it to create a dns record.
+    openshiftRouteOptions:
+      routerName: default
+    #Source Type is route resource of OpenShift
+    fqdnTemplate:
+    - "{{.Name}}.apps.test-azure1.qe.azure.devcluster.openshift.com"
+    type: OpenShiftRoute
+  zones:
+  - "/subscriptions/53b4f551-f0fc-4bea-8cba-11111111111/resourceGroups/test-azure1-nxkxm-rg/providers/Microsoft.Network/dnszones/test-azure1.qe.azure.devcluster.openshift.com"
+```
+
+9. Create this CR using 
+```bash
+oc create -f <cr-file-name>.yaml
+```
+
+10. Now you shall see records getting created for OCP created routes using the following command -
+```bash
+az network dns record-set list -g <resourcegroup>  -z <zone name you created in step 7 eg - misalunk-azure.qe.azure.devcluster.openshift.com>  > record-list-azure.txt
+```
+
+11. You can try to create a route with a sample app.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: hello-openshift
+  name: hello-openshift
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: hello-openshift
+  template:
+    metadata:
+      labels:
+        app: hello-openshift
+    spec:
+      containers:
+      - image: openshift/hello-openshift
+        name: hello-openshift
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: hello-openshift
+  name: hello-openshift
+spec:
+  ports:
+  - port: 8080
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: hello-openshift
+  sessionAffinity: None
+  type: ClusterIP
+---
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: hello-openshift
+spec:
+  host: hello-openshift.apps.<your domain, check your routes for reference>
+  to:
+    kind: Service
+    name: hello-openshift
+    weight: 100
+  wildcardPolicy: None
+
+```
+
+12. Create this route from the above file
+```bash
+$ oc create -f <resources>.yaml
+```
+
+13. Get the records for the dns zone
+```bash
+az network dns record-set list -g <resourcegroup>  -z <zone name you created in step 7 eg - misalunk-azure.qe.azure.devcluster.openshift.com>  > record-list-azure.txt
+```
+
+14.  Search for "hello" in the record-list-azure.txt and it shall exist.
```

Comment:
```suggestion
13. `hello-openshift` DNS records shall exist in the zone:
```bash
$ az network dns record-set list -g "${RESOURCE_GROUP}" -z <zone name you created in step 7 eg - misalunk-azure.qe.azure.devcluster.openshift.com>  | grep -c "hello-openshift"
```

14.  Search for "hello" in the record-list-azure.txt and it shall exist.
```